### PR TITLE
Update localization docs - guide & overview page

### PR DESCRIPTION
**Pull request by bc-andreadao**
## What/Why?

[DEVDOCS-6062](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6062)

This PR includes how to use subpaths that point to different channels for localization purposes. 

- Split Localization doc into 2 docs (overview & guide)
- Add overview page for localization use cases & URL structures 
- Add configuration for mapping locale to channel to the guide

This PR does NOT include how to use subpaths for language translations within the same channel, which is coming later.

Note: PR for directory structure:  https://github.com/bigcommerce/catalyst.dev/pull/18 

## Testing

N / A

[DEVDOCS-6062]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6062?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PAPI-2346]: https://bigcommercecloud.atlassian.net/browse/PAPI-2346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/localization-overview.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ 
+To fully localize a store for a language or region, you will need to customize product catalog and storefront content in the BigCommerce control panel.
+
+## Locale-specific URLs
+
+Catalyst supports the following locale-specific URLs that you can use to localize your storefronts. Learn more about the pro and cons for each [locale-specific URL](https://developers.google.com/search/docs/specialty/international/managing-multi-regional-sites#locale-specific-urls) in the Google Documentation.
+
+Currently, each Catalyst storefront can only support a single language. To display multiple languages, we recommend setting up a separate channel for each language.
+
+| URL structure | Example | Recommended use case | How to implement for Catalyst |
```
Comment (bc-andreadao) [inline docs/localization-overview.md:24]: @bookernath Not sure if I should have the "Recommended use case" column until after we support translated language content within the same channel? To my understanding, until then, all URL structures point to different channels (all URL structures could work equally well for Multi-GEO or Multi-Lang)

**Review by jorgemoya — APPROVED** [review]

File: docs/localization-overview.md (hunk 1 above)
Comment (bookernath) [reply docs/localization-overview.md]: I think this is fine for now and we can iterate on it later.

File: docs/localization/overview.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+2. You are selling within a country would like to accommodate multiple languages within that country
+ 
+To fully localize a store for a language or region, you will need to customize product catalog and storefront content in the BigCommerce control panel.
+
+## Locale-specific URLs
+
+Catalyst supports the following locale-specific URLs that you can use to localize your storefronts. Learn more about the pro and cons for each [locale-specific URL](https://developers.google.com/search/docs/specialty/international/managing-multi-regional-sites#locale-specific-urls) in the Google Documentation.
+
+Currently, each Catalyst storefront can only support a single language. To display multiple languages, we recommend setting up a separate channel for each language.
```
Comment (bookernath) [inline docs/localization/overview.md:22]:
```suggestion
Each Catalyst storefront can only support a single language per channel, but can service multiple language through the avenue of multiple channels. To display multiple languages, we recommend setting up a separate channel for each language.
```

File: docs/localization/overview.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+2. You are selling within a country would like to accommodate multiple languages within that country
+ 
+To fully localize a store for a language or region, you will need to customize product catalog and storefront content in the BigCommerce control panel.
+
+## Locale-specific URLs
+
+Catalyst supports the following locale-specific URLs that you can use to localize your storefronts. Learn more about the pro and cons for each [locale-specific URL](https://developers.google.com/search/docs/specialty/international/managing-multi-regional-sites#locale-specific-urls) in the Google Documentation.
+
+Each Catalyst storefront can support one language per channel. However, Catalyst can service multiple languages by connecting multiple channels to the storefront and routing each language through different subpaths. To display multiple languages, we recommend setting up ag separate channel for each lanuage.
```
Comment (bookernath) [inline docs/localization/overview.md:22]:
```suggestion
Each Catalyst storefront can support one language per channel. However, Catalyst can service multiple languages by connecting multiple channels to the storefront and routing each language through different subpaths. To display multiple languages, we recommend setting up a separate channel for each lanuage.
```

**Pull request merged**